server/jms/src/main/java/org/hawkular/apm/server/jms/BulkProcessingMDB.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.server.jms;
+
+import java.util.List;
+
+/**
+ * This MDB represents a bulk processor of the list of supplied items.
+ *
+ * @author gbrown
+ */
+public abstract class BulkProcessingMDB<T> extends RetryCapableMDB<T> {
+
+    /**
+    * {@inheritDoc}
+    */
+    @Override
+    protected void process(String tenantId, List<T> items, int retryCount) throws Exception {
+        try {
+            bulkProcess(tenantId, items, retryCount);
+        } catch (Exception e) {
```

Comment:
Shouldn't this catch something more specific than an exception? Maybe just the `ElasticSearchFailures`? I'd argue that there is no point in retrying the payloads if the fail on NPEs or some such.